JobRecruiting/JobRecruiting/src/main/java/com/recruit/jobrecruiting/user/ejb/UserBean.java

```diff
@@ -0,0 +1,167 @@
+        user.setFirstName(firstName);
+        user.setLastName(lastName);
+        user.setAddress(address);
+        
+        em.persist(user);
+    }
+    
+    public User getUserById(Integer id){
+        return em.find(User.class, id);
+    }
+    
+    public void activateUser(Integer id){
+        User user = getUserById(id);
+        user.setStatus(Status.ACTIVE);
+        
+        em.persist(user);
+    }
+    
+    public void deactivateUser(Integer id){
+        User user = getUserById(id);
+        user.setStatus(Status.INACTIVE);
+        
+        em.persist(user);
+    }
+    
+    public void updateUser(Integer id, String username, String email, String password, LocalDate birthDate, String firstName, String lastName, String address){
+        User user = getUserById(id);
+        user.setAddress(address);
+        user.setBirthDate(birthDate);
+        user.setFirstName(firstName);
+        user.setLastName(lastName);
+        user.setEmail(email);
+        user.setUsername(username);
+        user.setPassword(password);
+    }
+    
+    public void addCv(Integer id, String fileName, String fileType, byte[] fileContent){
+        //add image type
+        Photo photo = new Photo();
+        photo.setFilename(fileName);
+        photo.setFileType(fileType);
+        photo.setFileContent(fileContent);
+        
+        User user = getUserById(id);
+        user.setCv(photo);
+        
+        photo.setUser(user);
+        em.persist(photo);
+    }
+    
+    public void addProfilePicture(Integer id, String fileName, String fileType, byte[] fileContent){
+        //add image type
+        Photo photo = new Photo();
+        photo.setFilename(fileName);
+        photo.setFileType(fileType);
+        photo.setFileContent(fileContent);
+        
+        User user = getUserById(id);
+        user.setProfilePhoto(photo);
+        
+        photo.setUser(user);
+        em.persist(photo);
+    }
+    
+    private List<UserDetails> copyUsersToDetails(List<User> users) {
+        List<UserDetails> detailsList = new ArrayList<>();
+        for(User user:users){
+            UserDetails userDetails = new UserDetails(user.getId(), user.getUsername(), user.getEmail(), user.getPassword(), user.getBirthDate(), user.getFirstName(), user.getLastName(), user.getAddress(), user.getProfilePhoto(), user.getCv(), user.getStatus(), user.getPosition(), user.getDepartment());
+            detailsList.add(userDetails);
+        }
+        return detailsList;
+    }
+    
+    private List<UserLightDetails> copyUsersToLightDetails(List<User> users) {
+        List<UserLightDetails> detailsList = new ArrayList<>();
+        for(User user:users){
+            UserLightDetails userDetails = new UserLightDetails(user.getId(), user.getUsername(), user.getEmail(), user.getFirstName(), user.getLastName(), user.getPosition(), user.getStatus());
+            detailsList.add(userDetails);
+        }
+        return detailsList;
+    }
+}
```

Comment:
de for-urile astea se ocupa helperul detachEntities, entitatea ta trebe doar sa implementeze interfata Detachable si sa returneze din metoda un obiect UserDetails